### ISSUE TITLE
Integrate action

### ISSUE THREAD

**Issue by marina-chibizova** [1712n/dni-website#14]
[Here](https://github.com/1712n/challenge/issues/96#issuecomment-1540179265) is a solution that we need to integrate on this [website](https://dn.institute/). The solution is good, but I would appreciate your optimizations (e.g. he is using his personal token instead of a bot)

**Comment by danisztls** [comment]
Cool task and elegant solution!

The script can be integrated in the CI workflow as a pre-build action.

The tokens are fine, they can be stored as Github [secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) and read from the runtime environment.

I will play with his solution locally and give further feedback.

I suggest requesting [GPT-4](https://openai.com/product/gpt-4) access for this as the model is more accurate, up to date and can work with a significantly larger context.

I'm curious if we will be training embeddings in crypto corpora.

It's the 1st time I'm seeing LLMs being used for this application. Reminded me of [NewsMinimalist](https://www.newsminimalist.com/) (which has a different goal: relevance).

**Comment by marina-chibizova** [comment]
Thanks Daniel, I would appreciate your ideas!
@alxarno could you please also take a look from your perspective of actions best practices?

**Comment by marina-chibizova** [comment]
Notes from Alexey:
- Use runner's token for auth
- Job should fail in case fact check was not successful

**Comment by marina-chibizova** [comment]
@danisztls I agree with [Mikhail](https://github.com/1712n/challenge/issues/96#issuecomment-1541549916), no need to go for gpt-4 now. Could you please integrate it with [these](https://github.com/1712n/dni-website/issues/14#issuecomment-1540984241) fixes?

**Comment by marina-chibizova** [comment]
@danisztls do you have any ETA for this?

**Comment by danisztls** [comment]
This week at most.

**Comment by danisztls** [comment]
@marina-chibizova I'm improving the code in a private repo. To implement it here it will require:

- an [OpenAI key](https://platform.openai.com/) set as a secret with the name `OPENAI_KEY`
- a change in the repo setting so the Github runner automatic token have read/write permissions

Can you do that?

**Comment by marina-chibizova** [comment]
@danisztls done
